### PR TITLE
Add check-commits-count

### PR DESCRIPTION
**Pull request by enriquebelarte**
Add script to check that each PR consists of one single commit.

### CODE REVIEW

File: ci/prow/check-commits-count [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+number_of_commits=$(git rev-list --count HEAD ^main)
+if [[ ${number_of_commits} != 1 ]]; then
```
Comment (ybettan) [inline ci/prow/check-commits-count:6]: Did you test it? Why don't we check the last 2 commits like we do u/s?
https://github.com/kubernetes-sigs/kernel-module-management/pull/69/files

**Comment by ybettan** [comment]
Latest update for u/s code: https://github.com/kubernetes-sigs/kernel-module-management/pull/69

File: ci/prow/check-commits-count (hunk 1 above)
Comment (enriquebelarte) [reply ci/prow/check-commits-count]: You're right. Modified.

**Pull request merged**